src/NServiceBus.Core/EndpointCreator.cs

```diff
@@ -19,13 +19,33 @@ class EndpointCreator
             this.conventions = conventions;
         }
 
-        public static EndpointCreator Create(SettingsHolder settings, HostingComponent.Configuration hostingConfiguration)
+        public static EndpointCreator Create(EndpointConfiguration endpointConfiguration, IServiceCollection serviceCollection)
         {
-            var endpointCreator = new EndpointCreator(settings, hostingConfiguration, settings.Get<Conventions>());
+            var settings = endpointConfiguration.Settings;
+            CheckIfSettingsWhereUsedToCreateAnotherEndpoint(settings);
+
+            var assemblyScanningComponent = AssemblyScanningComponent.Initialize(settings.Get<AssemblyScanningComponent.Configuration>(), settings);
+
+            endpointConfiguration.FinalizeConfiguration(assemblyScanningComponent.AvailableTypes);
+
+            var hostingConfiguration = HostingComponent.PrepareConfiguration(settings.Get<HostingComponent.Settings>(), assemblyScanningComponent, serviceCollection);
 
+
+            var endpointCreator = new EndpointCreator(settings, hostingConfiguration, settings.Get<Conventions>());
             endpointCreator.Configure();
 
             return endpointCreator;
+
+            //TODO this would prevent calling Install/Start on the same endpoint config?
```

Comment:
should probably be addressed if possible. If it's technically not possible this would be a hard constraint on the API.